pkg/reconciler/service/occupancytracker.go

```diff
@@ -24,6 +25,7 @@ type OccupancyTracker struct {
 	occupancyID          string
 	occupancyCallbackURL string
 	ticker               *time.Ticker
+	sync.Mutex
```

Comment:
I think the mutex is never initialized in the new method

Reply:
Here, I am using struct and not a pointer to it, so it will be initialized to an unlocked Mutex state according to the docs.
`Mutex {0,0}`